lib/hocon/impl/simple_config_object.rb

```diff
@@ -107,6 +256,153 @@ def merged_with_object(abstract_fallback)
     end
   end
 
+  def modify(modifier)
+    begin
+      modify_may_throw(modifier)
+    rescue Hocon::ConfigError => e
```

Comment:
I'm fine with that. For the sake of consistency, though, we might want to change the instance of `RuntimeError` in this file to be a `ConfigError`.